File: week_3/homework/02_is_correct_parenthesis.py
s = "(())()"

def is_correct_parenthesis(string):
    check=0
    for s in string:
        if check<0:
            break
        elif s=='(':
            check+=1
        elif s==')':
            check-=1
        
    return True if check==0 else False

print(is_correct_parenthesis(s))  # True 를 반환해야 합니다!
print(is_correct_parenthesis("((((")) #False

File: week_3/homework/02_is_correct_parenthesis.cpp
#include<iostream>
#include<string>
using namespace std;

string st = "(())()";

bool is_correct_parenthesis(string str) {
	int check = 0;

	for (int i = 0; i < str.length(); i++) {
		if (check < 0)
			break;
		if (str[i] == '(')
			check++;
		else if (str[i] == ')')
			check--;
	}
	return check == 0 ? true : false;
}
int main(void) {
	bool ret=is_correct_parenthesis(st);

	cout << ret << '\n';
	return 0;
}